app/controllers/admin/content_controller.rb

```diff
@@ -37,6 +37,14 @@ def edit
     new_or_edit
   end
 
+  def merge
+    @article = Article.find(params[:id])
+    if current_user.admin?
+      @article.merge_with(params[:merge_with])
+    end
+    redirect_to :action => 'edit', id: @article.id
```

Comment:
The `merge_with` method in the model has some conditionals in place for bad merge requests. Since the merge form is already on the edit page, this redirect is pretty much just refreshing the page, whether the merge goes through or not. Is there something I'm not thinking of?